run_segmentation_training_pipeline.py

```diff
@@ -116,7 +116,10 @@ def time_str_to_seconds(timestr):
     print(f"Using dataset path {dataset_path}")
 
     # Get device
-    device = torch.device(args.device)
+    if args.device == 'cuda' and torch.cuda.is_available():
```

Comment:
Was thinking that if someone wants to use cpu they can pass --device cpu on the command line, but if GPU is not available and they don't set explicitly set CPU then it should raise an error.